test/test.ini

```diff
@@ -23,6 +23,9 @@ p62= port=6666 host=127.0.0.1 dbname=p6
 p7a= port=6666 host=127.0.0.1 dbname=p7
 p7b= port=6666 host=127.0.0.1 dbname=p7
 p7c= port=6666 host=127.0.0.1 dbname=p7
+p7x= port=6666 host=127.0.0.1 dbname=p7 min_pool_size=3 min_pool_size_requires_clients=0
+p7y= port=6666 host=127.0.0.1 dbname=p7 min_pool_size=3 min_pool_size_requires_clients=1 user=marko
+p7z= port=6666 host=127.0.0.1 dbname=p7 min_pool_size=3 min_pool_size_requires_clients=0 user=pswcheck
```

Comment:
The test relying on these works, but sadly this breaks tests that rely on the number of log `"new connection to server"` events

* `test/test_limits.py::test_min_pool_size_with_lower_max_db_connections` ([code](https://github.com/pgbouncer/pgbouncer/blob/bb018a6ac765b41d8d1c1129fe8c6ccc56018ec2/test/test_limits.py#L66-L68))
* `test/test_limits.py::test_min_pool_size_with_lower_max_user_connections` ([code](https://github.com/pgbouncer/pgbouncer/blob/bb018a6ac765b41d8d1c1129fe8c6ccc56018ec2/test/test_limits.py#L79-L81))
* `test/test_misc.py::test_host_list_dummy` ([code](https://github.com/pgbouncer/pgbouncer/blob/master/test/test_misc.py#L133-L134))

Suggestions on how to handle that?

Is there a way to define pool settings ad-hoc in a test (without having them in `test/test.ini`? That would avoid polluting the other tests.
Rewrite the other tests to be more specific about the connections? The log messages to contain the pool name and user name (e.g. `LOG S-0x562b6ab0c0a0: p0x/maxedout2@127.0.0.1:10211 new connection to server (from 127.0.0.1:58340)`)
Thoughts?

Reply:
there is the `run_with_config` context manager which allows you to specify a full config. Another option is to use the config line uncommenting trick that `test_auth_dbname_with_auto_database` uses.